### PR TITLE
Remove dead code paths and those not relating to ELT management

### PR DESCRIPTION
**Pull request by Ekrekr**
Folders removed are:
[dataform/redis](https://github.com/dataform-co/dataform/tree/main/redis)
[dataform/proto-gen-ts](https://github.com/dataform-co/dataform/tree/main/protoc-gen-ts)
[dataform/server](https://github.com/dataform-co/dataform/tree/main/server)
[dataform/sql](https://github.com/dataform-co/dataform/tree/main/sql)
[dataform/protomongo](https://github.com/dataform-co/dataform/tree/main/protomongo)
[dataform/grpc-web-proxy](https://github.com/dataform-co/dataform/tree/main/grpc-web-proxy)
[dataform/app](https://github.com/dataform-co/dataform/tree/main/app)
[dataform/components](https://github.com/dataform-co/dataform/tree/main/components)
[dataform/docs](https://github.com/dataform-co/dataform/tree/main/docs)

### CODE REVIEW

File: docs/BUILD [hunk 1]
```diff
@@ -1,34 +0,0 @@
-package(default_visibility = ["//visibility:public"])
```
Comment (BenBirt) [inline docs/BUILD:1]: isn't this used by the docs site?

File: examples/backwards_compatibility/BUILD [hunk 2]
```diff
@@ -1,19 +0,0 @@
-package(default_visibility = ["//tests:__subpackages__"])
```
Comment (BenBirt) [inline examples/backwards_compatibility/BUILD:1]: hm. are we sure we want to remove backwards compat testing? this effectively means we can unknowingly break backwards compatibility with old @dataform/core versions.

File: docs/BUILD (hunk 1 above)
Comment (lewish) [reply docs/BUILD:1]: It is, but on `main_v1` branch, so can delete safely here!

File: examples/backwards_compatibility/BUILD (hunk 2 above)
Comment (lewish) [reply examples/backwards_compatibility/BUILD:1]: +1 we should keep this

Comment (lewish) [reply examples/backwards_compatibility/BUILD:1]: I'd say remove the tests when we remove the support, but AIUI we still have it, and when we do remove it we should do a major version bump

Comment (Ekrekr) [reply examples/backwards_compatibility/BUILD:1]: SGTM

**Review by lewish — APPROVED** [review]

**Review by BenBirt — APPROVED** [review]

**Pull request merged**